lang/java/compiler/src/test/idl/input/simple.avdl

```diff
@@ -60,7 +60,7 @@ protocol Simple {
     time_ms t = 0;
 
     @foo.bar("bar.foo") long l = 0;
-    union {null, @foo.foo.bar(42) @foo.foo.foo("3foo") string} prop = null;
+    @foo.foo.bar(42) @foo.foo.foo("3foo") string? prop = null;
```

Comment:
This change demonstrates the new functionality. `simple.avpr` has not changed, as the result must be identical.

Reply:
IMO it would be good to add some comments next to the modified fields, for the future reader/you.
E.g. next to `time_ms? t = 0` we could add a comment like: `demonstrates a union schema with nonNull default value, so the nonNull schema is first`.
And next to `string? prop = null;` we could add a comment like: `demonstrates a union schema with null default value, so the nonNull schema is second`